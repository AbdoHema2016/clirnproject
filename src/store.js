import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
