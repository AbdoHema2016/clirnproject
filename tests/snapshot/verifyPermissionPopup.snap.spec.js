import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import VerificationPermissionUpdate from '../../src/containers/dashBoard/EmployerPermissions/verificationPopup';

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Verification permission', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Snapshot Pop up', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <VerificationPermissionUpdate {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
