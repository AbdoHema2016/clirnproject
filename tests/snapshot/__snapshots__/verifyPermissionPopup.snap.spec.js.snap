// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verification permission Snapshot Pop up 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(43,52,44,0.7)",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "paddingLeft": "5%",
        "width": "92%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "flexDirection": "row",
            "height": 48,
          },
          null,
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "height": 60.03,
            "marginTop": 8,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 4,
              "height": 52,
              "justifyContent": "center",
              "shadowColor": "#DDDDDD",
              "shadowOffset": Object {
                "height": 8,
                "width": 0,
              },
              "shadowOpacity": 0.75,
              "shadowRadius": 4,
            },
            Object {
              "height": "100%",
              "shadowColor": "rgba(0,0,0,0)",
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": undefined,
              "borderRadius": 4,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": 50,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "SofiaProRegular",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "height": 18,
                    "position": "absolute",
                    "right": 18,
                    "width": 18,
                  },
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ],
              ]
            }
          >
            <FastImageView
              resizeMode="contain"
              source={
                Object {
                  "cache": "immutable",
                  "uri": undefined,
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "alignSelf": "center",
            "height": 97,
            "width": 97,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "cache": "immutable",
            "uri": undefined,
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#565655",
            "fontFamily": "Futura",
            "fontSize": 24,
            "marginTop": "15%",
          },
        ]
      }
    >
      [missing "en.EMPLOYER_PERMISSIONS.VERIFICATION_POPUP_TITLE" translation]
    </Text>
    <Text
      style={
        Object {
          "marginRight": 5,
          "marginVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8D8C91",
            "fontFamily": "arial",
            "fontSize": 16,
            "fontWeight": "400",
            "marginBottom": "8%",
            "marginTop": "5%",
            "width": "90%",
          }
        }
      >
        [missing "en.EMPLOYER_PERMISSIONS.YOUR_ADMINISTRATOR" translation]
      </Text>
      <Text
        style={
          Object {
            "color": "#02BF8E",
            "fontFamily": "arial",
            "fontSize": 16,
            "fontWeight": "700",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#8D8C91",
            "fontFamily": "arial",
            "fontSize": 16,
            "fontWeight": "400",
            "marginBottom": "8%",
            "marginTop": "5%",
            "width": "90%",
          }
        }
      >
        [missing "en.EMPLOYER_PERMISSIONS.MODIFIED_PERMISSIONS" translation]
      </Text>
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": "2%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8D8C91",
              "fontFamily": "arial",
              "fontSize": 16,
              "fontWeight": "400",
            },
          ]
        }
      >
        [missing "en.EMPLOYER_PERMISSIONS.YOU_CAN" translation]
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#02BF8E",
              "fontFamily": "arial",
              "fontSize": 16,
              "fontWeight": "700",
              "width": "60%",
            },
          ]
        }
      >
        [missing "en.EMPLOYER_PERMISSIONS.APPROVE_MEDICAL_DATA" translation]
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#02BF8E",
            "marginBottom": "8%",
            "marginTop": "10%",
            "width": "95%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Futura",
                "fontSize": 16,
                "height": "40%",
              },
            ]
          }
        >
          [missing "en.EMPLOYER_PERMISSIONS.OK_MSG" translation]
        </Text>
      </View>
    </View>
  </View>
</View>
`;
